File: 520-detect-capital/detect-capital.py
class Solution(object):
    def detectCapitalUse(self, word):
        cap=0
        sm=0
        for i in range(len(word)):
            ch=word[i]
            if ch>='A' and ch<='Z':
                cap+=1
            else:
                sm+=1
        a=word[0]
        if cap==len(word):
            return True
        elif sm==len(word):
            return True
        elif a>='A' and a<='Z' and sm==len(word)-1:
            return True
        else:
            return False
        

File: 520-detect-capital/detect-capital.java
import java.util.*;
class Solution {
    public boolean detectCapitalUse(String word) {
        int cap=0;
        int sm=0;
        for(int i=0;i<word.length();i++){
            
            char ch = word.charAt(i);
            if(Character.isUpperCase(ch)){
                cap++;
            }
            else{
                sm++;
            }          
        }
        if(cap==word.length()) return true;
        if(sm==word.length())return true;
        if(Character.isUpperCase(word.charAt(0)) && sm==word.length()-1)return true;
        else return false;
    }
}